demo/admin/src/products/categories/AssignedProductsGrid.tsx

```diff
@@ -0,0 +1,57 @@
+import { CancelButton, SaveBoundary, SaveBoundarySaveButton } from "@comet/admin";
+import { Add as AddIcon } from "@comet/admin-icons";
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from "@mui/material";
+import { AssignProductsGrid } from "@src/products/categories/AssignProductsGrid";
+import { ProductsGrid } from "@src/products/ProductsGrid";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+type Props = {
+    productCategoryId: string;
+};
+
+export function AssignedProductsGrid({ productCategoryId }: Props): React.ReactElement {
+    const [isOpen, setIsOpen] = React.useState(false);
+
+    const handleCloseDialog = () => {
+        setIsOpen(false);
+    };
+    return (
+        <>
+            <ProductsGrid
+                hideCrudContextMenu={true}
+                toolbarAction={
+                    <Button startIcon={<AddIcon />} onClick={() => setIsOpen(true)} variant="contained" color="primary">
+                        <FormattedMessage id="products.editProductCategory.assignProducts" defaultMessage="Assign Products" />
+                    </Button>
+                }
+                filter={{ category: { equal: productCategoryId } }}
+            />
+            <SaveBoundary
+                onAfterSave={() => {
+                    setIsOpen(false);
+                }}
+            >
+                <Dialog open={isOpen} onClose={handleCloseDialog} fullWidth maxWidth="xl">
+                    <DialogTitle>
+                        <FormattedMessage id="products.editProductCategory.assignProducts" defaultMessage="Assign Products" />
+                    </DialogTitle>
+                    <DialogContent
+                        sx={{
+                            height: "70vh",
+                            padding: 0,
+                            paddingTop: "0 !important" /* is connected to title-style */,
+                        }}
+                    >
+                        <AssignProductsGrid productCategoryId={productCategoryId} />
+                    </DialogContent>
+                    <DialogActions>
+                        {/* TODO Missing close-dialog-unsaved-changes-check */}
```

Comment:
that should be working correclty, <Savable hasChanges is true it should work

Reply:
I checked again, it does not trigger a popup. I currently don't know why... probably because of the subRoutePath? opening the dialog does not change the path/url. it's still the same as for the routerTab.

Reply:
I had a look: you are using a plain mui Dialog. That obviously doesn't integrate our SaveBoundary.

You have to use our EditDialog instead

Reply:
ok, I started with comet-EditDialog but it had some issues with RouterTabs and StackPage so @thomasdax98 recommended to use plain mui Dialog. (probably you missed it: https://vividplanet.slack.com/archives/CTBMBLF38/p1718879902959449)